.github/workflows/build.yml

```diff
@@ -42,14 +42,18 @@ jobs:
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Run image
+        uses: abatilo/actions-poetry@v2.0.0
+        with:
+          poetry-version: "1.1.4"
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip tox
+          poetry install
       - name: Run tests
         if: ${{ !matrix.with-postgres }}
         run: |
-          tox -e py${{ matrix.python-version }}-django-${{ matrix.django-version}}
+          poetry run tox -e py${{ matrix.python-version }}-django-${{ matrix.django-version}}
       - name: Run tests on postgres
         if: ${{ matrix.with-postgres }}
         run: |
-          tox -e py${{ matrix.python-version }}-django-${{ matrix.django-version}} -- -p
+          poetry run tox -e py${{ matrix.python-version }}-django-${{ matrix.django-version}} -- -p
```

Comment:
You should pick whether poetry runs tox, or tox runs poetry. I think tox running poetry is probably the way to go.

What you don't want is to install the dependencies with poetry, then run tox to have it install all the dependencies with poetry again in the environment that it creates.